Introduce OSPP (Open Source Promotion Plan) (#184)
rust-vmm/community: ospp/README.md

after selection, carry out development under the guidance of project developers
(community mentors). Based on the project's difficulty and completion,
participants (*mentors and mentees*) will receive a bonus, and a completion
certificate for mentees.

# Who can Mentor
---
Regular contributors, maintainers and gatekeepers of Rust-VMM can participate
as mentors. Mentorship involves about 5 hours of time commitment per week to
communicate with the intern, review their patches, etc. Time is also required
during the intern selection phase to communicate with applicants. Being a
mentor is an opportunity to help someone get started in open source
development, will give you experience with managing a project in a low-stakes
environment, and a chance to explore interesting technical ideas that you may
not have time to develop yourself.

Please read mentor guide: https://summer-ospp.ac.cn/help/en/mentor/

How to Participate
---
As an organization, RustVMM need a `community contact` who is responsible for
registering RustVMM org in OSPP, and submit passed proposals to OSPP.

As a mentor, you should submit (raise a PR) a project proposal in:

> https://github.com/rust-vmm/community, if your PR gets merged, then your

Author: Good point 👍

(community mentors). Based on the project's difficulty and completion,
participants (*mentors and mentees*) will receive a bonus, and a completion
certificate for mentees.

# Who can Mentor
---
Regular contributors, maintainers and gatekeepers of Rust-VMM can participate
as mentors. Mentorship involves about 5 hours of time commitment per week to
communicate with the intern, review their patches, etc. Time is also required
during the intern selection phase to communicate with applicants. Being a
mentor is an opportunity to help someone get started in open source
development, will give you experience with managing a project in a low-stakes
environment, and a chance to explore interesting technical ideas that you may
not have time to develop yourself.

Please read mentor guide:
https://blog-en.summer-ospp.ac.cn/archives/mentor-guide

# How to Participate
---
As an organization, RustVMM need a `community contact` who is responsible for
registering RustVMM org in OSPP, and submit passed proposals to OSPP.

As a mentor, you should submit (raise a PR) a project proposal in:
https://github.com/rust-vmm/community, if your PR gets merged, then your
proposal is considered as accepted.

> ## Project proposal format

Reviewer: Should we mention where this proposal should be added?

Author: Noted, I will update it

```markdown
# Project Name

## Project Mentor

## Project Description

## Project Notes

## Project Difficulty

<Basic/Advanced>

## Project Size

> <N> Hours

Reviewer: GSoC provides fixed durations, is it the same here? If so, better to put the various choices here.

Author: No, it's left for mentors to decide

> ## Technical Domain Labels

Reviewer: Do we have some examples here?

project development tasks for the development and maintenance of important open
source software, and is open for registration to university students worldwide.
Students can independently choose projects they are interested in, apply, and
after selection, carry out development under the guidance of project developers
(community mentors). Based on the project's difficulty and completion,
**mentees** will receive a bonus, and a completion certificate.

# Who can Mentor

Regular contributors, maintainers and gatekeepers of rust-vmm can participate
as mentors. Mentorship involves about 5 hours of time commitment per week to
communicate with the intern, review their patches, etc. Time is also required
during the intern selection phase to communicate with applicants. Being a
mentor is an opportunity to help someone get started in open source
development, will give you experience with managing a project in a low-stakes
environment, and a chance to explore interesting technical ideas that you may
not have time to develop yourself.

Please read mentor guide:
https://blog-en.summer-ospp.ac.cn/archives/mentor-guide

# How to Participate

> As an organization, rust-vmm need a `community contact` who is responsible for

Reviewer: @RuoqingHe should we add that you are our community contact?

Author: Sure, I will update it 👍

# About Open Source Promotion Plan

The Open Source Promotion Plan is a summer program organized by the Open Source
Software Supply Chain Promotion Plan of the Institute of Software Chinese
Academy of Sciences from 2020. It aims to encourage university students to
actively participate in the development and maintenance of open source
software, cultivate and discover more outstanding developers, promote the
vigorous development of excellent open source software communities, and assist
in the construction of open source software supply chains.

The program collaborates with various open source communities, providing
project development tasks for the development and maintenance of important open
source software, and is open for registration to university students worldwide.
Students can independently choose projects they are interested in, apply, and
after selection, carry out development under the guidance of project developers
(community mentors). Based on the project's difficulty and completion,
participants (*mentors and mentees*) will receive a bonus, and a completion
certificate for mentees.

# Who can Mentor

> Regular contributors, maintainers and gatekeepers of Rust-VMM can participate

Reviewer: ```suggestion
Regular contributors, maintainers and gatekeepers of rust-vmm community can participate
```

as mentors. Mentorship involves about 5 hours of time commitment per week to
communicate with the intern, review their patches, etc. Time is also required
during the intern selection phase to communicate with applicants. Being a
mentor is an opportunity to help someone get started in open source
development, will give you experience with managing a project in a low-stakes
environment, and a chance to explore interesting technical ideas that you may
not have time to develop yourself.

Please read mentor guide:
https://blog-en.summer-ospp.ac.cn/archives/mentor-guide

# How to Participate

> As an organization, RustVMM need a `community contact` who is responsible for

Reviewer: ```suggestion
As an organization, rust-vmm need a `community contact` who is responsible for
```

> As an organization, RustVMM need a `community contact` who is responsible for

Reviewer: As I mentioned, I can be the `community contact` but I'd open another PR so we can agree on that.

(community mentors). Based on the project's difficulty and completion,
participants (*mentors and mentees*) will receive a bonus, and a completion
certificate for mentees.

# Who can Mentor

Regular contributors, maintainers and gatekeepers of Rust-VMM can participate
as mentors. Mentorship involves about 5 hours of time commitment per week to
communicate with the intern, review their patches, etc. Time is also required
during the intern selection phase to communicate with applicants. Being a
mentor is an opportunity to help someone get started in open source
development, will give you experience with managing a project in a low-stakes
environment, and a chance to explore interesting technical ideas that you may
not have time to develop yourself.

Please read mentor guide:
https://blog-en.summer-ospp.ac.cn/archives/mentor-guide

# How to Participate

As an organization, RustVMM need a `community contact` who is responsible for
registering RustVMM org in OSPP, and submit passed proposals to OSPP.

As a mentor, you should submit (raise a PR) a project proposal in:
https://github.com/rust-vmm/community, if your PR gets merged, then your
proposal is considered as accepted.

> For drafting a proposal, you may use the following template:

Reviewer: ```suggestion
For drafting a proposal, you may use the following template to add your proposal in the `ospp/OSPP-<year>.md` file:
```

## Project Proposal Template

```markdown
# Project Name

## Project Mentor

## Project Description

## Project Notes

## Project Difficulty

<Basic/Advanced>

## Project Size

<N> Hours

## Technical Domain Labels

[Cloud Native] | [Virtualization] | [Emulation] | ...

## Programming Language Labels

## Expect Outcome

## Project Technical Requirements

## Project Repository (Repo address of outcome is going to land)

```

> And you should raise a PR to modify `ospp/OSPP-<year>.md`.

Reviewer: ```suggestion
```
